internal/provider/resource_bytes.go

```diff
@@ -0,0 +1,169 @@
+}
+
+// Delete does not need to explicitly call resp.State.RemoveResource() as this is automatically handled by the
+// [framework](https://github.com/hashicorp/terraform-plugin-framework/pull/301).
+func (r *bytesResource) Delete(context.Context, resource.DeleteRequest, *resource.DeleteResponse) {
+}
+
+func (r *bytesResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	bytes, err := base64.StdEncoding.DecodeString(req.ID)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Import Random bytes Error",
+			"There was an error during the parsing of the base64 string.\n\n"+
+				diagnostics.RetryMsg+
+				fmt.Sprintf("Original Error: %s", err),
+		)
+		return
+	}
+
+	var state bytesModelV0
+
+	state.ID = types.StringValue("none")
+	state.Length = types.Int64Value(int64(len(bytes)))
+	state.ResultBase64 = types.StringValue(base64.StdEncoding.EncodeToString(bytes))
+	state.ResultHex = types.StringValue(hex.EncodeToString(bytes))
+	state.Keepers = types.MapValueMust(types.StringType, nil)
+
+	diags := resp.State.Set(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+type bytesModelV0 struct {
+	ID           types.String `tfsdk:"id"`
+	Length       types.Int64  `tfsdk:"length"`
+	Keepers      types.Map    `tfsdk:"keepers"`
+	ResultBase64 types.String `tfsdk:"result_base64"`
+	ResultHex    types.String `tfsdk:"result_hex"`
+}
+
+func bytesSchemaV3() tfsdk.Schema {
+	return tfsdk.Schema{
+		Version: 3,
+		Description: "The resource `random_bytes` generates random bytes that are intended to be " +
+			"used as secret or keys.",
+		Attributes: map[string]tfsdk.Attribute{
+			"keepers": {
+				Description: "Arbitrary map of values that, when changed, will trigger recreation of " +
+					"resource. See [the main provider documentation](../index.html) for more information.",
+				Type: types.MapType{
+					ElemType: types.StringType,
+				},
+				Optional: true,
+				PlanModifiers: []tfsdk.AttributePlanModifier{
+					resource.RequiresReplace(),
+				},
+			},
+			"length": {
+				Description: "The number of bytes requested. The minimum value for length is 1.",
+				Type:        types.Int64Type,
+				Required:    true,
+				PlanModifiers: []tfsdk.AttributePlanModifier{
+					resource.RequiresReplace(),
+				},
+				Validators: []tfsdk.AttributeValidator{
+					int64validator.AtLeast(1),
+				},
+			},
+			"result_base64": {
+				Description: "The generated bytes presented in base64 string format.",
+				Type:        types.StringType,
+				Computed:    true,
+				Sensitive:   true,
+			},
+			"result_hex": {
+				Description: "The generated bytes presented in hex string format.",
+				Type:        types.StringType,
+				Computed:    true,
+				Sensitive:   true,
+			},
+			"id": {
+				Description: "The generated bytes presented in string format.",
```

Comment:
at a cursory glance I don't think this is accurate, "id" appears to always be set to "none". Other providers have this described as "A static value used internally by Terraform, this should not be referenced in configurations."

Should `result` be added to actually return the generated bytes in string format? probably a question for @bendbennett - not sure how the state file is expected to behave with binary data in it - or whether it could be achieved in the `Read` function - or if anyone would ever use it anyway

Reply:
I did not set `result` to avoid confusion, since the result is always encoded. But I could add it, we just need to choose if the default is base64 or hex (or other ?)

Reply:
As the base64 and hex encodings of `result` are available it doesn't seem like there is a need for `result` as @Socolin indicates it would need to be encoded.